gnucash/import-export/import-main-matcher.c

```diff
@@ -138,11 +145,45 @@ static void refresh_model_row (
                     GNCImportTransInfo *info);
 /* end local prototypes */
 
+static gboolean delete_hash (gpointer key, gpointer value, gpointer user_data)
+{
+    // Value is a hash table that needs to be destroyed. 
+    g_hash_table_destroy (value);
+    return TRUE;
+}
+
+static void
+commit_all (GNCImportMainMatcher *info)
+{
+    // Commit the edits for all accounts for which we've called inc_account_edit.
+    for (;info->edited_accounts; info->edited_accounts=info->edited_accounts->next)
+    {
+        int level = qof_instance_get_editlevel (info->edited_accounts->data);
+        g_assert(level == 1);
+        xaccAccountCommitEdit (info->edited_accounts->data);
+    }
+    g_slist_free (info->edited_accounts);
```

Comment:
That won't work, `info->edited_accounts` is `NULL` now. You need an iterator in the `for` loop:
```C
for(GSList* iter = info->edited_accounts; iter; iter=iter->next)
```
and use `iter` instead of `info->edited_accounts` inside the loop.